drivers/usb/device/Kconfig

```diff
@@ -142,6 +142,22 @@ config USB_MCUX
 	help
 	  NXP MCUX USB Device Controller Driver for MXRT and LPC SoC's.
 
+config USB_MCUX_ENDPOINT_BUFFER_SIZE
+	int "Size of intermediate buffer in USB driver"
+	default 512
+	range 64 1024
+	help
+	  Size of intermediate buffer used in USB transfers. This is
+	  required for reads to ensure there is some area to receive
+	  the data. It is also required for writes when some descriptors
+	  to transfer are located in some area that could cause suffer
+	  from missed deadlines when the descriptor is in some area that
+	  has high latency to the USB DMA. The size will be used for a
+	  buffer for each endpoint. It is not recommended to change this
+	  unless the maximum size of all USB descriptors is known in advance.
```

Comment:
I really doubt it is descriptor issue, descriptors are placed in RAM section.

Reply:
I may be misunderstanding something but in this case it was `hid_report_desc` in the [HID sample](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/subsys/usb/hid/src/main.c), which was put into flash memory because it is declared const, which is why there was a flexspi performance issue and led to this PR.